src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs

```diff
@@ -947,7 +947,7 @@ private static ulong CurlReceiveHeadersCallback(IntPtr buffer, ulong size, ulong
                                 {
                                     response.Content.Headers.TryAddWithoutValidation(headerName, headerValue);
                                 }
-                                else if (easy._isRedirect && string.Equals(headerName, HttpKnownHeaderNames.Location, StringComparison.OrdinalIgnoreCase))
+                                else if (string.Equals(headerName, HttpKnownHeaderNames.Location, StringComparison.OrdinalIgnoreCase))
```

Comment:
I was looking to see what the HTTP 1.1 spec had to say about this sort of case.  In most things which talk about Location it is a "go here", but a HTTP 305 uses Location for "use this proxy".

https://tools.ietf.org/html/rfc2616#section-10.3.6.

So we should either still opt-in the redirecting codes, opt-out the 305, or decide "proxies are effectively dead, so who cares?"